python/tests/test_writer.py

```diff
@@ -496,3 +496,90 @@ def test_writer_with_options(tmp_path: pathlib.Path):
     )
 
     assert table == data
+
+
+def test_partition_overwrite(tmp_path: pathlib.Path):
```

Comment:
Could we parameterize this test by the partition column datatype? At very least, we should test `date32`, `timestamp`, and `bool`. I just want to verify that the data check function you wrote can handle those different types.

Reply:
You are correct on timestamp. I forgot about that. Created #1111 to track that.